standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdatePart.java

```diff
@@ -344,14 +281,10 @@ private void updateWriteIdForPartitions(Connection dbConn, long writeId, List<Lo
     TxnUtils.buildQueryWithINClause(conf, queries, prefix, suffix,
             partIdList, "\"PART_ID\"", false, false);
 
-    Statement statement = null;
     for (String query : queries) {
-      try {
-        statement = dbConn.createStatement();
-        LOG.debug("Going to execute update " + query);
+      LOG.debug("Going to execute update " + query);
+      try (Statement statement = dbConn.createStatement()) {
```

Comment:
why do we need new statements for each query? it looks like a batch update

Reply:
Addressed. Statement can be reused in this class.